server/src/test/java/umm3601/todo/TodoControllerSpec.java

```diff
@@ -0,0 +1,144 @@
+package umm3601.todo;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+
+import io.javalin.core.validation.Validator;
+import io.javalin.http.BadRequestResponse;
+import io.javalin.http.Context;
+import io.javalin.http.NotFoundResponse;
+
+import umm3601.Server;
+
+/**
+ * Tests the logic of the TodoController
+ *
+ * @throws IOException
+ */
+public class TodoControllerSpec {
+
+  private Context ctx = mock(Context.class);
+
+  private TodoController TodoController;
+  private static TodoDatabase db;
+
+  @BeforeEach
+  public void setUp() throws IOException {
+    ctx.clearCookieStore();
+
+    db = new TodoDatabase(Server.TODO_DATA_FILE);
+    TodoController = new TodoController(db);
+  }
+
+  @Test
+  public void GET_to_request_all_todos() throws IOException {
```

Comment:
Is this naming following a particular convention for naming? If not, I would suggest something more like test_GET_request_for_all_todos() or testGETrequestForAllTodos() - something that explains that this tests a get request that was asking for all the todos. I'm not sure about the 'to' in the name